hack/table-gen/table-gen.go

```diff
@@ -0,0 +1,264 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	REPattern      = `(?s)<!--\s*TABLE-START\s* -->.*<!--\s*TABLE-END\s*-->` //Regular expression patter for reading everything between TABLE-START and TABLE-END tags
+	SkipIdentifier = `SKIP-ELEMENT`
+	RESkipPattern  = `<!--\s*` + SkipIdentifier + `\s*([^\s]+)\s*-->` // Regular expression pattern for skipping one element without children that was specified in MD
+	// TODO: change to skip with children
+	SkipWithAncestorsIdentifier = `SKIP-WITH-ANCESTORS`
+	RESkipWithAncestorsPattern  = `<!--\s*` + SkipWithAncestorsIdentifier + `\s*([^\s-]+)\s*-->` // Regular expression pattern for skipping an element with its children
+)
+
```

Comment:
Are we using the skipping feature in any of our CRDs? I’m wondering if we will ever want to NOT generate a documentation for one of the fields in the CRDs. If this was already discussed, I’m fine with keeping it, but just curious to know.

Also, if we decide to have this skipping feature, then we should document how to use it with examples. Otherwise, it will be a hidden feature that you can only know about if you go through the source code. Maybe we create a README in the `table-gen` directory?

Reply:
This skipping feature may be necessary for other teams as well as for us. For instance, currently, we hardcoded skipping of the root `spec` and `status` elements, since we consider those elements should never be included. Meanwhile, other teams may need other elements be skipped as well, so we decided to keep this functionality. 

I agree on a README file, will add it as a resolution to this conversation